### PR TITLE
Comply with altitude for VTOL_LAND mission item

### PR DESCRIPTION
**Pull request by sanderux**
fixes https://github.com/PX4/Firmware/issues/4784

### CODE REVIEW

**Comment by sanderux** [comment]
@LorenzMeier AFAIK ready to merge

File: src/modules/navigator/mission.cpp [hunk 1]
```diff
@@ -519,13 +519,17 @@ Mission::set_mission_items()
 			/* use current mission item as next position item */
 			memcpy(&mission_item_next_position, &_mission_item, sizeof(struct mission_item_s));
 			has_next_position_item = true;
-			float altitude = _navigator->get_global_position()->alt;
-			if (pos_sp_triplet->current.valid) {
-				altitude = pos_sp_triplet->current.alt;
+
+			/* hold current altitude when no altitude is given */
+			if (_mission_item.altitude < 0.5f) {
```
Comment (LorenzMeier) [inline src/modules/navigator/mission.cpp]: So if I fly at sea level an altitude request of 0 meters will be rejected? This doesn't look completely thought out.

**Comment by LorenzMeier** [comment]
Doesn't look ready. It changes the logic entirely and it makes assumptions about magic values.

**Comment by sanderux** [comment]
The magic value is in QGC, it defaults the mission_item altitude to 0.
But you're right, it should check if it is different from 0, not just below zero.
or do you think we should ignore the mission_item's altitude all together?

**Comment by sanderux** [comment]
@LorenzMeier i addressed your comment, can you review bearing in mind that QGC defaults the altitude parameter to 0 on a VTOL_LAND command?

**Comment by sanderux** [comment]
@LorenzMeier is this ready to merge?

**Comment by dagar** [comment]
Would it make more sense to change the default in QGC?

**Comment by dagar** [comment]
It seems like MAV_CMD_NAV_VTOL_LAND needs a separate transition altitude and landing altitude.

**Comment by sanderux** [comment]
@LorenzMeier can you review this again? (note that stable QGC sets altitude to 0 by default for VTOL_LAND)

**Comment by dagar** [comment]
@sanderux does it make sense now that VTOL_LAND specifies the landing coordinates (lat, lon, alt) just like regular LAND? The QGC default is 0 relative to home. Could we add an additional optional param to VTOL_LAND that also specifics the back transition altitude?

**Comment by sanderux** [comment]
I think we need a different discussion on the meaning of altitude in a land command as it does not seem to perform what i would expect.

**Comment by sanderux** [comment]
closing in favor of changes pending from https://github.com/PX4/Firmware/issues/5824

**Pull request closed without merging**